BondageClub/Scripts/Activity.js

```diff
@@ -82,75 +82,200 @@ function ActivityDictionaryText(KeyWord) {
+ */
+function ActivityCheckPrerequisite(prereq, acting, acted, group) {
+	switch (prereq) {
+		case "UseMouth":
+			return (!acting.IsMouthBlocked() && acting.CanTalk());
+		case "UseTongue":
+			return !acting.IsMouthBlocked();
+		case "TargetMouthBlocked":
+			return acted.IsMouthBlocked();
+		case "IsGagged":
+			return !acting.CanTalk();
+		case "SelfOnly":
+			return (acted.ID == 0);
+		case "TargetKneeling":
+			return acted.IsKneeling();
+		case "UseHands":
+			return acting.CanInteract();
+		case "UseArms":
+			return (acting.CanInteract()
+				&& (InventoryGet(acting, "ItemArms") == null || !InventoryGroupIsBlocked(acting, "ItemArms")));
+		case "UseFeet":
+			return acting.CanWalk();
+		case "CantUseArms":
+			return !(acting.CanInteract()
+				&& (InventoryGet(acting, "ItemArms") == null || !InventoryGroupIsBlocked(acting, "ItemArms")));
+		case "CantUseFeet":
+			return !acting.CanWalk();
+		case "TargetCanUseTongue":
+			return !acted.IsMouthBlocked();
+		case "TargetMouthOpen":
+			if (group.Name === "ItemMouth")
+				return (InventoryGet(acted, "ItemMouth") && acted.IsMouthOpen());
+			break;
+		case "VulvaEmpty":
+			if (group.Name === "ItemVulva")
+				return !acted.IsVulvaFull();
+			break;
+		case "MoveHead":
+			if (group.Name === "ItemHead")
+				return !acted.IsFixedHead();
+			break;
+		case "ZoneAccessible":
+			return !InventoryGroupIsBlocked(acted, group.Name, true);
+		case "WearingPenetrationItem":
+			return CharacterHasItemForActivity(acting, "Penetrate") && !acting.IsEnclose();
+		case "ZoneNaked":
+			if (group.Name === "ItemButt")
+				return (InventoryPrerequisiteMessage(acted, "AccessButt") === "" && !acted.IsPlugged());
+			else if (group.Name === "ItemVulva")
+				return ((InventoryPrerequisiteMessage(acted, "AccessVulva") === "") && !acted.IsVulvaChaste());
+			else if (group.Name === "ItemBreast" || group.Name === "ItemNipples")
+				return ((InventoryPrerequisiteMessage(acted, "AccessBreast") === "") && !acted.IsBreastChaste());
+			else if (group.Name === "ItemBoots")
+				return (InventoryPrerequisiteMessage(acted, "NakedFeet") === "");
+			else if (group.Name === "ItemHands")
+				return (InventoryPrerequisiteMessage(acted, "NakedHands") === "");
+			break;
+	}
+	return true;
+}
+
+/**
+ * Check that an activity's prerequisites are met.
+ * @param {Activity} activity - The activity to consider
+ * @param {Character|PlayerCharacter} acting - The character performing the activity
+ * @param {Character|PlayerCharacter} acted - The character being acted on
+ * @param {AssetGroup} group - The group being acted on
+ * @returns {boolean} whether the given activity's prerequisite are satisfied
+ */
+function ActivityCheckPrerequisites(activity, acting, acted, group) {
+	if (!activity.Prerequisite)
+		return true;
 
+	for (const pre of activity.Prerequisite) {
+		if (!ActivityCheckPrerequisite(pre, acting, acted, group))
+			return false;
 	}
+	return true;
```

Comment:
One more minor thing now that this has been refactored, but it would probably be cleaner to use `Array.prototype.every`:
```suggestion
	return activity.Prerequisite.every((pre) => ActivityCheckPrerequisite(pre, acting, acted, group));
```